modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java

```diff
@@ -920,162 +922,170 @@ protected ServiceProvider generateServiceProviderWithRoles(OAuthConsumerAppDTO a
-				});
-				restClient.addAPIAuthorizationToApplication(applicationId, authorizedAPICreationModel);
-			} catch (Exception e) {
-				throw new RuntimeException("Error while authorizing system API " + apiIdentifier + " to application "
-						+ applicationId, e);
-			}
-		});
-	}
+        JSONParser parser = new JSONParser();
+        JSONObject json = (JSONObject) parser.parse(EntityUtils.toString(response.getEntity()));
+        return ((JSONArray) ((JSONObject) ((JSONArray) json.get("keys")).get(0)).get("x5c")).get(0).toString();
+    }
 
-	public String getRoleV2ResourceId(String roleName, String audienceType, String OrganizationId) throws Exception {
+    /**
+     * Authorize list of SYSTEM APIs to an application.
+     *
+     * @param applicationId  Application id.
+     * @param apiIdentifiers API identifiers to authorize.
+     * @throws Exception Error occured while authorizing APIs.
+     */
+    public void authorizeSystemAPIs(String applicationId, List<String> apiIdentifiers) throws Exception {
+
+        apiIdentifiers.stream().forEach(apiIdentifier -> {
+            try {
+                List<APIResourceListItem> filteredAPIResource =
+                        restClient.getAPIResourcesWithFiltering("identifier+eq+" + apiIdentifier);
+                if (filteredAPIResource == null || filteredAPIResource.isEmpty()) {
+                    return;
+                }
+                String apiId = filteredAPIResource.get(0).getId();
+                // Get API scopes.
+                List<ScopeGetModel> apiResourceScopes = restClient.getAPIResourceScopes(apiId);
+                AuthorizedAPICreationModel authorizedAPICreationModel = new AuthorizedAPICreationModel();
+                authorizedAPICreationModel.setId(apiId);
+                authorizedAPICreationModel.setPolicyIdentifier("RBAC");
+                apiResourceScopes.forEach(scope -> {
+                    authorizedAPICreationModel.addScopesItem(scope.getName());
+                });
+                restClient.addAPIAuthorizationToApplication(applicationId, authorizedAPICreationModel);
+            } catch (Exception e) {
+                throw new RuntimeException("Error while authorizing system API " + apiIdentifier + " to application "
+                        + applicationId, e);
+            }
+        });
+    }
 
-		List<String> roles = restClient.getRoles(roleName, audienceType, OrganizationId);
-		if (roles.size() == 1) {
-			return roles.get(0);
-		}
-		return null;
-	}
+    public String createBusinessAPIs(BusinessAPICreationModel businessAPICreationModel) {
+        try {
+            return restClient.createBusinessAPIResource(businessAPICreationModel);
+        } catch (Exception e) {
+            throw new RuntimeException("Error while creating business API " + businessAPICreationModel.getName());
+        }
+    }
+
+    public void authorizeBusinessAPIs(String applicationId, AuthorizedAPICreationModel authorizedAPICreationModel) {
```

Comment:
Let's rename as authorizeDomainAPIs

Reply:
Fixed with the commit 6e4aab9c2f07c6fa7ac9e7d13b2ea705224a25a5